From 1dceab6c75ce7bad309da9114d9e7f518291a955 Mon Sep 17 00:00:00 2001
From: Shawn Routhier <sar@isc.org>
Date: Wed, 10 Sep 2014 20:26:40 -0700
Subject: [PATCH] [master] Remove unneeded defines for errors

As part of 4.2.0 we renamed a number of errors but continued
to include the old ones in case somebody was building something
using our libraries.  This #ifdefs them and by defaults removes them
from the compile.
---
diff --git a/includes/omapip/result.h b/includes/omapip/result.h
index 3e4cf6a..ae5f7d6 100644
--- a/includes/omapip/result.h
+++ b/includes/omapip/result.h
@@ -80,13 +80,13 @@
 
 // Included for historical reasons, these should be removed as
 // soon as reasonable
+#ifdef INCLUDE_OLD_DHCP_ISC_ERROR_CODES
 #define ISC_R_HOSTUNKNOWN	DHCP_R_HOSTUNKNOWN
 #define ISC_R_VERSIONMISMATCH	DHCP_R_VERSIONMISMATCH
 #define ISC_R_PROTOCOLERROR	DHCP_R_PROTOCOLERROR	
 #define ISC_R_INVALIDARG	DHCP_R_INVALIDARG	
 #define ISC_R_NOTYET		DHCP_R_NOTYET		
 #define ISC_R_UNCHANGED		DHCP_R_UNCHANGED		
-#define ISC_R_MULTIPLE		DHCP_R_MULTIPLE
 #define ISC_R_KEYCONFLICT	DHCP_R_KEYCONFLICT	
 #define ISC_R_BADPARSE		DHCP_R_BADPARSE		
 #define ISC_R_NOKEYS		DHCP_R_NOKEYS		
@@ -113,6 +113,7 @@
 #define ISC_R_NOT_EQUAL		DHCP_R_NOT_EQUAL		
 #define ISC_R_CONNRESET		DHCP_R_CONNRESET		
 #define ISC_R_UNKNOWNATTRIBUTE	DHCP_R_UNKNOWNATTRIBUTE	
+#endif
 
 isc_result_t
 dhcp_result_register(void);
diff --git a/includes/site.h b/includes/site.h
index d87b309..19a2e11 100644
--- a/includes/site.h
+++ b/includes/site.h
@@ -304,6 +304,14 @@
    removal of this define.  Use at your own risk.  */
 /* #define ENABLE_GENTLE_SHUTDOWN */
 
+/* Include old error codes.  This is provided in case you
+   are building an external program similar to omshell for
+   which you need the ISC_R_* error codes.  You should switch
+   to DHCP_R_* error codes for those that have been defined
+   (see includes/omapip/result.h).  The extra defines and
+   this option will be removed at some time. */
+/* #define INCLUDE_OLD_DHCP_ISC_ERROR_CODES */
+
 /* Include definitions for various options.  In general these
    should be left as is, but if you have already defined one
    of these and prefer your definition you can comment the 
-- 
2.1.0

